nautobot/ipam/views.py

```diff
@@ -550,21 +538,21 @@ def get_extra_context(self, request, instance):
             .restrict(request.user, "view")
             .select_related("role", "status", "tenant")
             .prefetch_related("primary_ip4_for", "primary_ip6_for")
-            .annotate(
-                interface_count=count_related(Interface, "ip_addresses"),
-                interface_parent_count=count_related(Device, "interfaces__ip_addresses", distinct=True),
-                vm_interface_count=count_related(VMInterface, "ip_addresses"),
-                vm_interface_parent_count=count_related(VirtualMachine, "interfaces__ip_addresses", distinct=True),
-            )
         )
 
         # Add available IP addresses to the table if requested
         if request.GET.get("show_available", "true") == "true":
-            ipaddresses = add_available_ipaddresses(
-                instance.prefix, ipaddresses, instance.type == choices.PrefixTypeChoices.TYPE_POOL
-            )
 
-        ip_table = tables.IPAddressTable(ipaddresses)
+            def data_transform_callback(ip_addresses):
+                return add_available_ipaddresses(
+                    instance.prefix, ip_addresses, instance.type == choices.PrefixTypeChoices.TYPE_POOL
+                )
+        else:
+            data_transform_callback = None
```

Comment:
Or since you already refactoring this "pattern" maybe go even further and prepare more robust functions like:
```
def add_available_ipaddresses_callback(instance, show_available):
    ...

data_transform_callback = add_available_ipaddresses_callback(instance, request.GET.get("show_available", "true")
```
I can see multiple benefits: we get rid off weird `def` from the view, remove this `if request.`, move callback logic into one place, allow simpler unittest for each callback function. And for the `PrefixPrefixes` also remove additional queryset for `child_prefixes` from the get extra context.